sdks/python/apache_beam/io/parquetio.py

```diff
@@ -64,14 +64,24 @@
 class _ArrowTableToRowDictionaries(DoFn):
   """ A DoFn that consumes an Arrow table and yields a python dictionary for
   each row in the table."""
-  def process(self, table):
+  def process(self, table, with_context=False):
+    if with_context:
+      path = table[0]
+      table = table[1]
     num_rows = table.num_rows
     data_items = table.to_pydict().items()
-    for n in range(num_rows):
-      row = {}
-      for column, values in data_items:
-        row[column] = values[n]
-      yield row
+    if with_context:
+      for n in range(num_rows):
```

Comment:
We should rearrange the code so that we don't copy blocks for code.

Reply:
I refactored this, please let me know if that would work. Thanks